packages/kbn-reporting/public/share/share_context_menu/register_pdf_png_modal_reporting.tsx

```diff
@@ -0,0 +1,519 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ShareContext, ShareMenuItem, ShareMenuProvider } from '@kbn/share-plugin/public';
+import { FormattedMessage, InjectedIntl } from '@kbn/i18n-react';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import { checkLicense } from '../../license_check';
+import {
+  ExportModalShareOpts,
+  ExportPanelShareOpts,
+  JobParamsProviderOptions,
+  ReportingSharingData,
+} from '.';
+import { ScreenCapturePanelContent } from './screen_capture_panel_content_lazy';
+import { ReportingAPIClient } from '../../reporting_api_client';
+
+const getJobParams =
+  (
+    apiClient: ReportingAPIClient,
+    opts: JobParamsProviderOptions,
+    type: 'pngV2' | 'printablePdfV2'
+  ) =>
+  () => {
+    const {
+      objectType,
+      sharingData: { title, layout, locatorParams },
+    } = opts;
+
+    const baseParams = {
+      objectType,
+      layout,
+      title,
+    };
+
+    if (type === 'printablePdfV2') {
+      // multi locator for PDF V2
+      return { ...baseParams, locatorParams: [locatorParams] };
+    }
+    // single locator for PNG V2
+    return { ...baseParams, locatorParams };
+  };
```

Comment:
`apiClient` isn't used
```suggestion
const getJobParams = (opts: JobParamsProviderOptions, type: 'pngV2' | 'printablePdfV2') => () => {
  const {
    objectType,
    sharingData: { title, layout, locatorParams },
  } = opts;

  const baseParams = {
    objectType,
    layout,
    title,
  };

  if (type === 'printablePdfV2') {
    // multi locator for PDF V2
    return { ...baseParams, locatorParams: [locatorParams] };
  }
  // single locator for PNG V2
  return { ...baseParams, locatorParams };
};
```